collectors/freebsd.plugin/freebsd_sysctl.c

```diff
@@ -1077,9 +1077,9 @@ int do_system_ram(int update_every, usec_t dt) {
         else rrdset_next(st_mem_available);
 
 #if __FreeBSD_version < 1200016
-        rrddim_set_by_pointer(st_mem_available, rd_avail, vmmeter_data.v_inactive_count + vmmeter_data.v_free_count + (vmmeter_data.v_cache_count * system_pagesize + zfs_arcstats_shrinkable_cache_size_bytes));
+        rrddim_set_by_pointer(st_mem_available, rd_avail, vmmeter_data.v_inactive_count + vmmeter_data.v_free_count + vmmeter_data.v_cache_count + zfs_arcstats_shrinkable_cache_size_bytes / system_pagesize);
 #else
-        rrddim_set_by_pointer(st_mem_available, rd_avail, vmmeter_data.v_inactive_count + vmmeter_data.v_free_count + zfs_arcstats_shrinkable_cache_size_bytes);
+        rrddim_set_by_pointer(st_mem_available, rd_avail, vmmeter_data.v_inactive_count + vmmeter_data.v_free_count + zfs_arcstats_shrinkable_cache_size_bytes / system_pagesize);
```

Comment:
@MrZammler @vlvkobal @thiagoftsm check if we need to case to a different type here, I don't know.

- vmmeter_data.* is `unsigned int`
- zfs_arcstats_shrinkable_cache_size_bytes is `long long int`
- system_pagesize is `int`

Reply:
Hello @ilyam8 ,

In theory we are working with `integers` here, but you address important points:

- The biggest difference for `unsigned int` and `int` it is the way one bit is interpreted, it is a number for the first and a signal for the second.
- Now when we talk about `long long int` and `int` they have size difference between them.

I would suggest to convert everything to `long long int` to be sure that nothing would be lost. I would not expect that all compilers will require you to cast the data, `gcc 12`, for example, does not say nothing.

Best regards!